mlrun/datastore/sources.py

```diff
@@ -812,6 +812,11 @@ def __init__(
             attributes["sasl"] = sasl
         super().__init__(attributes=attributes, **kwargs)
 
+    def to_dataframe(self):
+        raise mlrun.MLRunInvalidArgumentError(
+            "KafkaSource does not support batch processing"
```

Comment:
suggest to guide the user explicitly to ingestion service:
Stream sources (like KafkaSource) are not supported by local feature-set ingest, use deploy_ingestion_service()  instead